### PR TITLE
Add multi-tab for code in SetupModal

### PR DESCRIPTION
**Pull request by hypervillain**
[Linear ticket](https://linear.app/prismic/issue/SM-917/aauser-i-can-see-multitab-to-code-block-in-setupmodal-in-slice)

- Add tabs for npm and yarn in Setup modal
- Optimize tabs to prevent unnecessary reload

To test it:
- Remove prop `localSimulatorURL` from `sm.json`
- Access the simulator
- Check the first Setup tab
- alternate between yarn and npm: both code and copy should match
- the other code blocks should still match previous design

How it looks
Capture d’écran 2023-01-03 à 17 29 00

### CODE REVIEW

File: packages/slice-machine/components/Simulator/components/SetupModal/index.tsx [hunk 1]
```diff
@@ -88,7 +89,30 @@ const SetupModal: React.FC<{ isOpen: boolean }> = ({ isOpen }) => {
...
+          >
+            <Step
+              stepNumber={i + 1}
+              linkToStorybookDocs={linkToStorybookDocs}
+              linkToTroubleshootingDocs={linkToTroubleshootingDocs}
+            />
+          </TabPanel>
+        );
+      }),
+    [setupData]
```
Comment (mathieud) [inline packages/slice-machine/components/Simulator/components/SetupModal/index.tsx]: Could you add the missing dependencies?

File: packages/slice-machine/components/Simulator/components/SetupModal/VersionBadgeList.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+const VersionBadgeList: FC<{
+  versions: string[];
+  selectedVersion: string;
+  setSelectedVersion: (s: string) => void;
+}> = ({ versions, selectedVersion, setSelectedVersion }) => (
+  <Flex sx={{ maxWidth: "fit-content", p: "16px 16px 0 16px" }}>
+    {versions.map((version, i) => (
+      <VersionBadge
+        key={version}
+        sx={versions[i + 1] ? { mr: "12px" } : undefined}
```
Comment (mathieud) [inline packages/slice-machine/components/Simulator/components/SetupModal/VersionBadgeList.tsx]: We could have this on the `Flex` parent, avoiding the condition

```
'& > *': {
  mr: "12px",
}
```

File: packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx [hunk 3]
```diff
@@ -5,9 +5,16 @@ import { MdCheck, MdContentCopy } from "react-icons/md";
...
+import VersionBadgeList from "./VersionBadgeList";
+
+type Code = {
+  text: string;
+  version: string;
+};
 
 const CodeBlockWithCopy: React.FC<{
-  children: string;
+  children?: string;
```
Comment (mathieud) [inline packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx]: I'm not a fan of having two possible source. I'd rather have `children: string | Code[]` and deal with it in the component. It feels cleaner, makes the interface simpler and easier to understand and user errors less likely (here could be called with neither children nor codelist)

**Review by mathieud — COMMENTED** [review]
Do we need the changes to the `package.lock`?

File: packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx (hunk 3 above)
Comment (hypervillain) [reply packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx]: Yeah me either. But I'm not a fan of children being anything but a node. I'll find something else

File: packages/slice-machine/components/Simulator/components/SetupModal/index.tsx (hunk 1 above)
Comment (hypervillain) [reply packages/slice-machine/components/Simulator/components/SetupModal/index.tsx]: ✅

File: packages/slice-machine/components/Simulator/components/SetupModal/VersionBadgeList.tsx (hunk 2 above)
Comment (hypervillain) [reply packages/slice-machine/components/Simulator/components/SetupModal/VersionBadgeList.tsx]: ✅

File: packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx (hunk 3 above)
Comment (hypervillain) [reply packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx]: Would it be ok for you if I removed `children` and only kept `code`, of type `Code | Code[]` ?

Comment (hypervillain) [reply packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx]: https://github.com/prismicio/slice-machine/pull/794/commits/88613f6af5ae95fc842bdbe565e46716e77ee802#diff-9d2f70d013cc7c10e6d41982d7fdd59904c6c12627482ffd9a9ae8546146024bR16

I updated, LMK!

Comment (mathieud) [reply packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx]: Yep really nice, exactly what I had in mind

File: packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx [hunk 4]
```diff
@@ -62,43 +65,58 @@ const CodeBlockWithCopy: React.FC<{
...
+              >
+                Copy&nbsp;
+                {isCopied ? (
+                  <MdCheck size={16} color={String(theme?.colors?.success)} />
+                ) : (
+                  <MdContentCopy size={16} style={{ marginLeft: "8px" }} />
+                )}
+              </Button>
+            </Flex>
+            {codeList?.length && codeList.length > 1 ? (
```
Comment (mathieud) [inline packages/slice-machine/components/Simulator/components/SetupModal/CodeBlock.tsx]: nit: `codeList` is always defined

File: packages/slice-machine/components/Simulator/components/SetupModal/VersionBadgeList.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  <Flex
+    sx={{
+      maxWidth: "fit-content",
+      p: "16px 16px 0 16px",
+      "& > button:not(:last-child)": {
+        mr: "12px",
+      },
+    }}
+  >
+    {versions.map((version, i) => (
```
Comment (mathieud) [inline packages/slice-machine/components/Simulator/components/SetupModal/VersionBadgeList.tsx]: `i` is not used anymore

**Pull request merged**